### ISSUE TITLE
Source Swap: No Audio

### ISSUE THREAD

**Issue by WarDogSniper** [wwmm/pulseeffects#144]
Using Pulse Effects Flatpak (current)
OS: Ubuntu-MATE 17.04 (up-to date)

2 or 3 updates ago this was not an issue but now any time I switch sources, either audio or video, pulse effects stops giving me output of my new source.  it doesn't lock up, the meters are still moving but no audio out. The only way to get it back is to close and re-open pulse effects again. 

ex.
listening to station-1 in mpv or vlc and switch to station-2 and I must close & re-open pulse effects to hear anything. Same with video 

The kicker here is that it seems to be only in mpv, vlc and the like and not in a browser. I can change video and audio streams without it cutting audio out. but once I change sources in mpv/vlc audio is dead for all audio out.
edit:
after updating and for some unknown reason it is now dropping audio out for browsers too. Another kicker is I don't even need to change sources, I just need to skip ahead in the same video and it drops out. 
My Favorite app has now become unusable.

**Comment by wwmm** [comment]
Hi. Are you able to run directly from the sources [Run-Directly-from-Repository](https://github.com/wwmm/pulseeffects/wiki/Run-Directly-from-Repository)? I do not have this problem here but I use the package I maintain for Arch Linux. When developing I run directly from the sources and there is no issue too. So it may be an issue specific to the flatpack package.

Please follow the instructions here [Reporting Bugs](https://github.com/wwmm/pulseeffects/wiki/Reporting-Bugs). And paste here the output before and after the problem happens

**Comment by hyphone** [comment]
Hello there

it doesn't work for me at all (Ubuntu 17.10).

Flathub-, repository- and the version directly from GIT source results in no audio at all.
I can close and re-open pulseeffects with no success.

Perhaps the wrong source is activated so I activated pulseeffects (apps) manually but also with no success. I just get audio again when I choose my normal Speaker / Headphone output.

Can this mean something?

`PulseEffects - WARNING - (gerror=GLib.Error('Der Ton kann nicht schnell genug aufgezeichnet werden', 'gst-core-error-quark', 13), debug="gstaudiobasesrc.c(847): gst_audio_base_src_create (): /GstPipeline:pipeline0/GstPulseSrc:audio_src:\nDropped 39690 samples. This is most likely because downstream can't keep up and is consuming samples too slowly.")
`

"Der Ton kann nicht schnell genug aufgezeichnet werden" means something like:
_"The sound could not be recorded fast enough"_

**Comment by wwmm** [comment]
Every now and then I have the message `The sound could not be recorded fast enough` too. Unless you have lots of this message you should not be worried. 

If you can see the levels and the spectrum but can not listen to audio the are too possibilities. Either PulseEffects is playing audio to an output device you do no expect it to be playing or it is recording from the wrong device. A few Pulseaudio's modules like `module-device-manager` and `module-switch-on-connect` are know to conflict with PulseEffects (issues that will be fixed in the next Pulseaudio release). But only looking at the logs we can know if they have anything to do with your problem

**Comment by wwmm** [comment]
@hyphone please follow the steps here [Reporting Bugs](https://github.com/wwmm/pulseeffects/wiki/Reporting-Bugs)

**Comment by wwmm** [comment]
@hyphone  I think that ubuntu 17.10 loads the module `module-switch-on-connect` by default. This will probably cause the problem you are facing. Until the new Pulseaudio is released you will have to unload this module before running PE `pactl unload-module module-switch-on-connect`

**Comment by hyphone** [comment]
@wwmm thank you and sorry for not giving a detailed report. yes that worked.
I guess Ubuntu users will have to wait until 18.04 or unload the said module.

**Comment by wwmm** [comment]
Another option would be updating to the new Pulseaudio when it is released. Maybe someone will do a ppa  with it.

**Comment by WarDogSniper** [comment]
>@wwmm
 Hi. Are you able to run directly from the sources Run-Directly-from-Repository? I do not have this problem here but I use the package I maintain for Arch Linux. When developing I run directly from the sources and there is no issue too. So it may be an issue specific to the flatpack package.

I was unable to successfully get this to run directly or compile/install following the readme file I just got missing directories and files errors. can't recall the errors right off hand. I rolled my box back before leaving town.

>Please follow the instructions in Reporting Bugs. And paste here the output before and after the problem happens

I will work on error reports as soon as soon as real life work slows down again. this is a busy time of year

**Comment by WarDogSniper** [comment]
I do believe I found the problem and it is not on your end.  In my case I use firejail and remembered at one time in the past there was an issue using firejail and pulse audio. I believe the bug was in the kernel and was eventually fixed to some degree and was no longer being talked about. Remembering this I went back in my notes and found the fix for pulse/firejail and everything seems to be back in working order once again.
Thanks for the Great app @wwmm!
